ori/volume/server/volume_create.go

```diff
@@ -16,261 +16,143 @@ package server
-		},
+	labels := volume.Metadata.Labels
+	if labels == nil {
+		labels = map[string]string{}
 	}
 
-	var imagePvc *corev1.PersistentVolumeClaim
-	if volume.Spec.Image != "" {
-		imageName := GetSanitizedImageName(volume.Spec.Image)
-		apiutils.SetImageAnnotation(pvc, volume.Spec.Image)
+	labels[VolumeClassLabel] = volume.Spec.Class
 
-		pvc.Spec.DataSourceRef = &corev1.TypedObjectReference{
-			APIGroup: pointer.String("snapshot.storage.k8s.io"),
-			Kind:     "VolumeSnapshot",
-			Name:     imageName,
+	var snapshotRef *string
+	var snapshot *api.Snapshot
+	if image := volume.Spec.Image; image != "" {
+		img, err := s.registry.Resolve(ctx, image)
+		if err != nil {
+			return nil, fmt.Errorf("failed to resolve image: %w", err)
 		}
 
-		size, err := getImageSize(ctx, volume.Spec.Image)
+		onmetalImage, err := onmetalimage.ResolveImage(ctx, img)
 		if err != nil {
-			return nil, fmt.Errorf("unable to get image size: %w", err)
+			return nil, fmt.Errorf("failed to resolve onmetal image: %w", err)
 		}
 
-		imagePvc = &corev1.PersistentVolumeClaim{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "PersistentVolumeClaim",
-				APIVersion: "v1",
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      imageName,
-				Namespace: s.namespace,
-			},
-			Spec: corev1.PersistentVolumeClaimSpec{
-				AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
-				Resources: corev1.ResourceRequirements{
-					Requests: corev1.ResourceList{corev1.ResourceStorage: *size},
-				},
-				VolumeMode:       func(m corev1.PersistentVolumeMode) *corev1.PersistentVolumeMode { return &m }(corev1.PersistentVolumeBlock),
-				StorageClassName: pointer.String(s.rookPoolStorageClass),
-				//set DataSourceRef that populator picks up the pvc
-				DataSourceRef: &corev1.TypedObjectReference{
-					APIGroup: pointer.String(storagev1alpha1.SchemeGroupVersion.String()),
-					Kind:     "Volume",
-					Name:     pvc.Name,
-				},
-			},
+		if onmetalImage.RootFS == nil {
+			return nil, fmt.Errorf("failed to get rootFS layer")
 		}
-		metautils.SetLabel(imagePvc, volumev1alpha1.ManagerLabel, volumev1alpha1.VolumeCephletManager)
-
-	}
-
-	if err := apiutils.SetObjectMetadata(pvc, volume.Metadata); err != nil {
-		return nil, err
-	}
-	metautils.SetLabel(pvc, volumev1alpha1.ManagerLabel, volumev1alpha1.VolumeCephletManager)
-	apiutils.SetVolumeClassLabel(pvc, volume.Spec.Class)
-
-	return &AggregateCephVolume{
-		Pvc:         pvc,
-		ImagePvc:    imagePvc,
-		VolumeClass: volumeClass,
-	}, nil
-}
 
-func (s *Server) handleImagePopulation(ctx context.Context, log logr.Logger, volume *AggregateCephVolume) error {
-	if volume.ImagePvc == nil {
-		return nil
-	}
-
-	snapshot := &snapshotv1.VolumeSnapshot{}
-	if err := s.client.Get(ctx, client.ObjectKeyFromObject(volume.ImagePvc), snapshot); err != nil {
-		if !errors.IsNotFound(err) {
-			return fmt.Errorf("unable to get snapshot: %w", err)
+		imageSize := uint64(onmetalImage.RootFS.Descriptor().Size)
```

Comment:
This shouldn't be checked at creation time but rather be done during a reconciliation.